Scripts and styles: Remove redundant HTML5 type attributes

sirreal opened the pull request:
Building on https://github.com/WordPress/wordpress-develop/pull/10645 (included here).

Trac ticket: https://core.trac.wordpress.org/ticket/64428

> The script-loader and WP_Styles both make checks against current_theme_supports( 'html5', 'style' ) and current_theme_supports( 'html5', 'script' ), and multiple places in the rest of the codebase unconditionally produce type="text/css" and type="text/javascript" HTML attributes on the associated tags.
> 
> This is a legacy convention predating HTML5 in 2008 and can be removed. These are the default values when the attributes are absent, and this does not depend on or change with declared theme support.
> 
> Removing the type attribute simplifies logic and normalizes the produced HTML content.
> 
> This is part of [#59883](https://core.trac.wordpress.org/ticket/59883)
> 
> See also [PR#10641](https://github.com/WordPress/wordpress-develop/pull/10641)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

hardikRathi committed:
Trac #64428: Remove default type attributes from SCRIPT and STYLE elements

hardikRathi committed:
Trac #64428: Restore CDATA condition and keep explicitly set script types.

sirreal committed:
Remove text/css type attribute from theme style tags

sirreal committed:
Remove text/css type attribute from theme link tags

sirreal committed:
Remove text/css type from style tags

sirreal committed:
Remove type-attr from class-wp-styles

sirreal committed:
Remove text/css type from class-wp-font-face

sirreal committed:
Remove text/css type from Core LINK tags

sirreal committed:
Update scripts tests

sirreal committed:
Update inline script tag tests

sirreal committed:
Update styles tests

sirreal committed:
Merge branch 'trunk' into 64428/remove-type-attr-script-style-tags

sirreal committed:
Update fonts tests without text/css type

sirreal committed:
Update privacy test without text/css

sirreal commented:
This has a lot of overlap with https://github.com/WordPress/wordpress-develop/pull/10660.

westonruter commented on src/wp-includes/deprecated.php:
```diff
@@ -5897,10 +5897,9 @@ function _wp_theme_json_webfonts_handler() {
 	_deprecated_function( __FUNCTION__, '6.4.0', 'wp_enqueue_embed_styles' );
 
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
 	$suffix    = SCRIPT_DEBUG ? '' : '.min';
```
```suggestion
	$suffix = SCRIPT_DEBUG ? '' : '.min';
```

westonruter reviewed (approved):
A thing of beauty. 

Just one little gnat to pluck out, but I'm pre-approving.

sirreal committed:
Fix equals alignment

Co-authored-by: Weston Ruter <westonruter@gmail.com>

sirreal commented on src/wp-includes/fonts/class-wp-font-face.php:
```diff
@@ -72,21 +72,6 @@ class WP_Font_Face {
-			&&
-			function_exists( 'current_theme_supports' ) && ! current_theme_supports( 'html5', 'style' )
-		) {
-			$this->style_tag_attrs = array( 'type' => 'text/css' );
```
This is a private property that never appears to be written to in this class, it can be removed.

sirreal commented on tests/phpunit/tests/dependencies/scripts.php:
```diff
@@ -186,7 +186,7 @@ public function test_after_inline_script_with_delayed_main_script( $strategy ) {
 	 *
 	 * If a main script with a `blocking` strategy has an `after` inline script,
-	 * the inline script should be rendered as type='text/javascript'.
+	 * the inline script should be rendered as.
```
I'm not sure what to do with this comment or whether it was a good description of the test. I'll remove it.

sirreal committed:
Merge branch 'trunk' into 64428/remove-type-attr-script-style-tags

sirreal committed:
Remove unused style_tag_attrs and generate_style_element_attributes

This property and method are private and only print an empty string.
They are safe to remove.

sirreal committed:
Removed out-of-date comment

sabernhardt commented on src/wp-includes/script-loader.php:
```diff
@@ -2987,7 +2969,7 @@ function wp_get_inline_script_tag( $data, $attributes = array() ) {
 	 */
 	if (
-		! $is_html5 &&
+		! current_theme_supports( 'html5', 'script' ) &&
```
The `$is_html5` variable checked for theme support **plus** `is_admin()`. With r61411, if I activate a theme such as Twenty Eleven, I now get `CDATA` for scripts in the admin.
```
<script id="utils-js-extra">
/* <![CDATA[ */
var userSettings = {"url":"/svn/src/","uid":"1","time":"1766767047","secure":""};
//# sourceURL=utils-js-extra
/* ]]> */
</script>
```

Of course, this should not need fixing if #10660 removes `CDATA` from _all_ scripts.

sirreal replied:
Thanks for spotting that!

I'm happy addressing it with a small fix: https://github.com/WordPress/wordpress-develop/pull/10666.

But I do expect https://github.com/WordPress/wordpress-develop/pull/10660 to remove all the CDATA wrappers soon.

sirreal replied:
Handled in r61414.
